validator/keymanager/remote-web3signer/keymanager.go

```diff
@@ -38,54 +53,323 @@ type SetupConfig struct {
 	// Either URL or keylist must be set.
 	// a static list of public keys to be passed by the user to determine what accounts should sign.
 	// This will provide a layer of safety against slashing if the web3signer is shared across validators.
-	ProvidedPublicKeys [][48]byte
+	ProvidedPublicKeys []string
 }
 
 // Keymanager defines the web3signer keymanager.
 type Keymanager struct {
 	client                internal.HttpSignerClient
 	genesisValidatorsRoot []byte
-	publicKeysURL         string
-	providedPublicKeys    [][48]byte
+	providedPublicKeys    [][48]byte          // source of truth flag loaded + file loaded + api loaded keys
+	flagLoadedKeysMap     map[string][48]byte // stores what was provided from flag as a map
 	accountsChangedFeed   *event.Feed
 	validator             *validator.Validate
-	publicKeysUrlCalled   bool
+	retriesRemaining      int
+	keyFilePath           string
+	lock                  sync.RWMutex
 }
 
 // NewKeymanager instantiates a new web3signer key manager.
-func NewKeymanager(_ context.Context, cfg *SetupConfig) (*Keymanager, error) {
+func NewKeymanager(ctx context.Context, cfg *SetupConfig) (*Keymanager, error) {
+	ctx, span := trace.StartSpan(ctx, "remoteKeymanager")
```

Comment:
I think this is the only span in the file. Do you want to add spans to other places as well? Also the span name is a bit unusual. It should be something like `remote-keymanager.NewKeymanager` or `keymanager.remote.NewKeymanager`.